airflow/jobs/local_task_job_runner.py

```diff
@@ -158,7 +158,7 @@ def sigusr2_debug_handler(signum, frame):
         try:
             self.task_runner.start()
 
-            heartbeat_time_limit = conf.getint("scheduler", "scheduler_zombie_task_threshold")
+            heartbeat_time_limit = conf.getint("scheduler", "local_task_job_heartbeat_sec")
```

Comment:
I wonder why a different heartbeat was used here instead of the jobs heartrate. Do we really need this configuration or we should use JOB_HEARTBEAT_SEC? It's already set for this localtaskjob, If there's no apparent reason for a different configuration, I think we should use self.job.heartrate for this without any other configuration

Reply:
Hey Ephraim! For us atleast, when we are running our jobs, we have these default values set and having these default values set differently for different job types is very convenient for us. Anyways the idea of this is also coming from https://github.com/apache/airflow/issues/30908. Definitely open to discuss however